### DISCUSSION TITLE
Sympy dev

### DISCUSSION THREAD

**Discussion by klimanek** [sympy/sympy#26987] [General]
Hello everyone,

I joined the Sympy community, like many others, not only to identify bugs but also to fix them. Unfortunately, I'm still struggling to grasp how certain things work, such as the calculations of integrals. I understand that heuristics are used first, followed by the Risch algorithm, etc. But what does this mean in practice? Is the heuristic approach implemented in `heurisch.py`? If I want to fix the integral calculation, where should I start? The only thing that is immediately understandable to me is `manualintegrate.py`.

I'll choose an even simpler example. Sympy doesn't handle the expression `0^t` correctly. Such a function is continuous only on `(0, oo)`, but Sympy states that it's continuous over the entire ℝ. (I described this problem here: https://github.com/sympy/sympy/issues/26949.)

Singularities and continuity here are only calculated from the denominators. If there's no denominator in the expression, Sympy assumes that no singularities can exist. It doesn't consider that a negative exponent expresses a fraction. This should be easy to fix, but where?

One last question: In VSCode, the code is filled with red river, both Pylance and Black are throwing errors. Should I disable everything locally, or is there a recommended configuration for code checks?

Thanks!
PS: I forgot to add the most important thing: some online workshops along with tut. videos on YT would be a great start for all newcommers.

**Comment by moorepants** [comment]
> One last question: In VSCode, the code is filled with red river, both Pylance and Black are throwing errors. Should I disable everything locally, or is there a recommended configuration for code checks?

SymPy code mostly adheres to the PEP8 standard and our documentation follows our style guide. Linting tools that are for other purposes will likely display warnings.

**Comment by oscarbenjamin** [comment]
> I'm still struggling to grasp how certain things work, such as the calculations of integrals.

The integration code is not an easy part of the codebase to work on. Feel free to go and read about the Risch algorithm, Meijer G functions etc but these are not simple algorithms and you need to do a lot of self learning if you want to be able to understand the code there.

> Sympy doesn't handle the expression `0^t` correctly. Such a function is continuous only on `(0, oo)`, but Sympy states that it's continuous over the entire ℝ.

When you say "SymPy" here what you mean is the `continuous_domain` function from the calculus module. That is a much easier part of the codebase to work on than the integration code and the best thing to do is just go read the code.

> In VSCode, the code is filled with red river, both Pylance and Black are throwing errors. Should I disable everything locally

Disable black and don't use any autoformatter on the sympy codebase. There is configuration for both ruff and flake8 so you could use either of those for basic checks.

Pylance is also problematic because it struggles on the sympy codebase: https://github.com/microsoft/pyright/issues/7159.

I have a PR that improves the situation for pyright/pylance and other downstream type checkers and autocompleters etc: gh-25103.
